### ISSUE TITLE
CelebA-collar -> CSR

### ISSUE THREAD

**Issue by joshua840** [joshua840/ConceptualSensitivityRegularization#5]
# Migrate CelebA-collar dataset into our repository.

We have reproduced the rrClarC method into our repository, but we cannot check whether it is well migrated. Also, we need a real-world experiment in our paper. So we decided to migrate this dataset into our repo. 
- [x] Copy the code and fit the form with our abstract class
- [x] Create ipynb file and check whether it is well implemented
---
## Workflow
- Create a new branch based on this issue (use vscode github extention)
- Continuously commit work done in the branch
- When doing pull-request, use git-rebase. It will help you follow the logs based on the name of issues

**Comment by joshua840** [comment]
```python
self.metadata = pd.DataFrame(
    {
        "image_id": np.array(ds.filename)[filter_indices == 1],
        "targets": labels,
        "blonde_collar": np.logical_and(labels, labels2) == 1,
    }
)
``` 
target: blond (1) / not blond (0)
spurious: collar (1) / not collar (0)

**Comment by joshua840** [comment]
tr val te split:

How to do this?

**Comment by joshua840** [comment]
we need to define minor_ratio of celeba_concept dataset

**Comment by joshua840** [comment]
minor_ratio = the ratio of minority sample over another group with the same class

zero, another one will be added

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/b5d48bfb-cd01-4246-90a2-28c7f710b1fb)
[0,1] and [1,1] is not saved..

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/70c644d8-0834-45fe-8d4a-d1b822c22e29)
raw data is containing [1,1]. Moreover, the number of target==1 is 2395+16, which is the same as in EpochChangeable~. I think something went wrong during saving features (treats [1,1] as [1,0])

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/41238757-e355-4952-bf93-31a073236507)
I found the reason that I passed 0, not g value.

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/754f97de-9e4d-457f-8a2a-6bb67c243d4e)
Do augmentation value is set to False. Is it okey? I think it is possible that the every features have the same value without augmentation.

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/2809d74d-4f48-42c0-a411-581350e9791a)

This is a sample of celebaCollar dataset. y represent the hair color, and g represent whether they have collar or not. 
The number of population with blonde+collar is only 16 in the training dataset.

**Comment by joshua840** [comment]
If a model classifies an image as non-blonde because the one is wearing a collar, the accuracy on the group blonde-collar will be very low. 
What we need is to let the model to not distinguish between blonde-collar and blonde-non-collar.

**Comment by joshua840** [comment]
![image](https://github.com/joshua840/ConceptualSensitivityRegularization/assets/24368050/d88cdb05-dce0-49fb-8934-e3f7ccc54b5e)
The current ConceptDataset is constructed with blonde/with-collar vs non_blonde/with-collar. In this case, the CAV will point out the hair collar direction. Regularizing this direction would have a harmful impact.

**Comment by joshua840** [comment]
```python
    def compute_cav(self, mode, norm=False):
        vecs = []
        sample_ids = []

        path = self.path
        classes = range(self.n_classes)
        if self.cav_scope == "attacked_class":
            if "celeba" in self.dataset_name:
                classes = [1]
            if "imagenet" in self.dataset_name:
                classes = [0]
            if "isic" in self.dataset_name:
                classes = [0]
            if "bone" in self.dataset_name:
                classes = [2]
        for class_id in classes:
            data = torch.load(f"{path}/{self.layer_name}_class_{class_id}_all.pth")
            if data['samples']:
                sample_ids += data['samples']
                vecs.append(torch.stack(data[mode], 0))
``` 
I've checked the code in rrClarC repo., showing that it only leverages classes=1 in the extracted features. I think they did well, and we need to change our CelebaCollar concept dataset.

**Comment by joshua840** [comment]
- How about leveraging already implemented information?
  - Simply load CelebaBiased from the rrClarC repo. 
  - Or extract the data names and save it in the yaml file.
  - You can choose both, but the result will be changed if you choose option B

**Comment by joshua840** [comment]
We created a new `.yaml` file that can be loaded from Concept dataset.

**Comment by joshua840** [comment]
Closing this issue as the PR has been merged.
